bridge/src/lib.rs

```diff
@@ -0,0 +1,61 @@
+#[macro_use]
+extern crate rutie;
+extern crate lazy_static;
+
+mod connection;
+
+use connection::Connection;
+use rutie::{Module, Object, Symbol, RString, Encoding, AnyObject, AnyException, Exception, VM, Thread};
+
+wrappable_struct!(Connection, ConnectionWrapper, CONNECTION_WRAPPER);
+
+class!(TemporalBridge);
+
+methods!(
+    TemporalBridge,
+    rtself,
+
+    fn create_connection(host: RString) -> AnyObject {
+        let host = host.map_err(|e| VM::raise_ex(e)).unwrap().to_string();
+
```

Comment:
This pattern with raising the error would be much better expressed by having these functions return a result, and then if the result is an error, raising then. (It seems like that should just be what happens, and you shouldn't have to explicitly raise an error -- worth checking if these return results if it ends up working that way).

So you could Move these functions outside of the `methods!` macro (which, is a good idea anyway since these types of macros often break IDE support). Then, this function becomes something like:

```rust
fn create_connection(host: RString) -> AnyObject {
    match create_connection_impl(host) {
        Err(e) => {
            VM::raise_ex(e);
            // I have no idea what you are supposed to return here. The code now will just panic which also seems wrong. Hence why I suspect we should be able to just return a result type.
        }
        Ok(r) => r
    }
}
```

Reply:
Ah, I see, basically add another layer of abstraction that handles Ruby errors while keeping the interface clean and return a `Result`. This makes lots of sense, thanks!

Would you mind if I do this in a follow-up PR along with the `thiserror` change?